src/platform/linux.c

```diff
@@ -252,9 +256,28 @@ int platform_open_common_file_dialog(filedialog_type type, utf8 *title, utf8 *fi
 					break;
 			}
 
+			// Zenity seems to be case sensitive, while Kdialog isn't.
 			if (filterPattern && filterName) {
-				filter = (char*) malloc(strlen("--file-filter=\"") + strlen(filterPattern) + 3 + strlen(filterName) + 2);
-				sprintf(filter, "--file-filter=\"%s | %s\"", filterName, filterPattern);
+				if(strcmp(filterPattern, "*.sv6") == 0)
+					filterPatternRegex = "*.[Ss][Vv]6";
+				else if(strcmp(filterPattern, "*.sc6") == 0)
+					filterPatternRegex = "*.[Ss][Cc]6";
+				else if(strcmp(filterPattern, ".td6") == 0)
+					filterPatternRegex = "*.[Tt][Dd]6";
+				else if(strcmp(filterPattern, "*.sv4") == 0)
+					filterPatternRegex = "*.[Ss][Vv]4";
+				else if(strcmp(filterPattern, "*.sc4") == 0)
+					filterPatternRegex = "*.[Ss][Cc]4";
+				else if(strcmp(filterPattern, "*.td4") == 0)
+					filterPatternRegex = "*.[Tt][Dd]4";
+				else if(strcmp(filterPattern, "*.dat") == 0)
+					filterPatternRegex = "*.[Dd][Aa][Tt]";
+				else
+					filterPatternRegex = (char *)filterPattern;
```

Comment:
I suggest changing this to automatically convert the pattern into case insensitive. Loop through the characters, add the brackets and add the lowercase / uppercase.